packages/aws-cdk/lib/util/npm.ts

```diff
@@ -0,0 +1,19 @@
+import { exec as _exec } from 'child_process';
+import { promisify } from 'util';
+import * as semver from 'semver';
+import { debug } from '../../lib/logging';
+
+const exec = promisify(_exec);
+
+export async function getLatestVersionFromNpm(): Promise<string> {
```

Comment:
I broke this method out into a separate file/module so I can easily stub it in tests. It's challenging/confusing to mock a method in the same file with Jest or Sinon (see https://stackoverflow.com/questions/45111198/how-to-mock-functions-in-the-same-module-using-jest and https://stackoverflow.com/questions/39861674/stubbing-method-in-same-file-using-sinon)